File: 01_Algorithmic_Toolbox/week1_programming_challenges/2_maximum_pairwise_product/maxpairwiseproduct.py
def MaxPairwiseProduct(num):
    """
    product = 0
    """
    n = len(num)
    """
    for i in range(n):
        for j in range(i + 1, n):
            product = max(product, num[i] * num[j])
    """
    num_sorted = sorted(num)
    product = num_sorted[-1] * num_sorted[-2]
    return product

if __name__ == "__main__":
    n = int(input())
    numbers = [int(x) for x in input().split()]

    print(MaxPairwiseProduct(numbers))

File: 01_Algorithmic_Toolbox/week1_programming_challenges/2_maximum_pairwise_product/maxpairwiseproduct.R
MaxPairwiseProduct <- function(num){
#    product <- 0
    n <- length(num)
#    for(i in 1:(n - 1)){
#        for(j in (i + 1):n){
#            product <- max(product, num[i] * num[j])
#        }
#    }
    num_sorted <- sort(num, decreasing = T)
    product <- num_sorted[1] * num_sorted[2]
    return(product)
}

n <- as.integer(readline(prompt=""))
numbers <- as.integer(unlist(strsplit(readline(prompt=""), " ")))

print(MaxPairwiseProduct(numbers))